src/ints/mouse_dos_driver.cpp

```diff
@@ -0,0 +1,1440 @@
+        state.maxpos_y = static_cast<int16_t>(8 * (rows + 1) - 1);
+        break;
+    }
+    case 0x04:
+    case 0x05:
+    case 0x06:
+    case 0x08:
+    case 0x09:
+    case 0x0a:
+    case 0x0d:
+    case 0x0e:
+    case 0x13:
+        if (mode == 0x0d || mode == 0x13)
+            state.gran_x = 0xfffe;
+        state.maxpos_y = 199;
+        break;
+    case 0x0f:
+    case 0x10: state.maxpos_y = 349; break;
+    case 0x11:
+    case 0x12: state.maxpos_y = 479; break;
+    default:
+        LOG(LOG_MOUSE, LOG_ERROR)
+        ("Unhandled videomode %X on reset", mode);
+        state.inhibit_draw = true;
+        return;
+    }
+
+    state.mode               = mode;
+    state.maxpos_x           = 639;
+    state.minpos_x           = 0;
+    state.minpos_y           = 0;
+    state.hot_x              = 0;
+    state.hot_y              = 0;
+    state.user_screen_mask   = false;
+    state.user_cursor_mask   = false;
+    state.text_and_mask      = DEFAULT_TEXT_AND_MASK;
+    state.text_xor_mask      = DEFAULT_TEXT_XOR_MASK;
+    state.page               = 0;
+    state.update_region_y[1] = -1; // offscreen
+    state.cursor_type        = MouseCursor::Software;
+    state.enabled            = true;
+
+    MOUSE_NotifyResetDOS();
+}
+
+static void Reset()
+{
+    MOUSEDOS_BeforeNewVideoMode();
+    MOUSEDOS_AfterNewVideoMode(false);
+
+    SetMickeyPixelRate(8, 16);
+    SetDoubleSpeedThreshold(0); // set default value
+
+    state.enabled    = true;
+    state.cute_mouse = false;
+
+    pos_x = static_cast<float>((state.maxpos_x + 1) / 2);
+    pos_y = static_cast<float>((state.maxpos_y + 1) / 2);
+
+    state.mickey_x = 0;
+    state.mickey_y = 0;
+
+    state.last_wheel_moved_x = 0;
+    state.last_wheel_moved_y = 0;
+
+    for (uint16_t idx = 0; idx < NUM_BUTTONS; idx++) {
+        state.times_pressed[idx]   = 0;
+        state.times_released[idx]  = 0;
+        state.last_pressed_x[idx]  = 0;
+        state.last_pressed_y[idx]  = 0;
+        state.last_released_x[idx] = 0;
+        state.last_released_y[idx] = 0;
+    }
+
+    state.sub_mask              = 0;
+    mouse_shared.dos_cb_running = false;
+
+    UpdateDriverActive();
+    MOUSE_NotifyResetDOS();
+}
+
+static void LimitCoordinates()
+{
+    auto limit = [](float &pos, const int16_t minpos, const int16_t maxpos) {
+        const float min = static_cast<float>(minpos);
+        const float max = static_cast<float>(maxpos);
+
+        pos = std::clamp(pos, min, max);
+    };
+
+    // TODO: If the pointer go out of limited coordinates,
+    //       trigger showing mouse_suggest_show
+
+    limit(pos_x, state.minpos_x, state.maxpos_x);
+    limit(pos_y, state.minpos_y, state.maxpos_y);
+}
+
+static void UpdateMickeysOnMove(float &dx, float &dy, const float x_rel,
+                                const float y_rel)
+{
+    auto calculate_d = [](const float rel,
+                          const float pixel_per_mickey,
+                          const float senv) {
```

Comment:
senv?

Reply:
Legacy code, I haven't analyzed it - I have just put into lambda the common part.